docs/pages/docs/apis/access-control.mdx

```diff
@@ -20,244 +20,171 @@ export default config({
-      // Declarative access control definition
-      access: { isLocked: { equals: false } },
+      access: {
+        operation: {
+          query: ({ session, context, listKey, operation }) => true,
+          create: ({ session, context, listKey, operation }) => true,
+          update: ({ session, context, listKey, operation }) => true,
+          delete: ({ session, context, listKey, operation }) => true,
+        }
+      },
     }),
   }),
 });
 ```
 
-Declarative access control cannot be used for the `create` operation, as there is no filter being applied when creating an item.
-If you use declarative access control with the `create` operation, or with concise syntax, it is equivalent to the static access control definition `true` for create operations.
-
-Declarative access control rules are rarely used directly.
-A more common pattern is to return a declarative access control rule from an [imperative](#imperative-list) rule.
+?> The `query` access control rule is applied when running GraphQL query operations.
+It does not prevent a user **reading** the data returned by the mutation operations.
 
-### Imperative (list)
+### Filter
 
-Imperative access control rules are functions which return either a boolean value or a [declarative](#declarative-list) value (i.e. a GraphQL `where` clause).
-Imperative access control functions can be either synchronous or async.
-The function is passed a set of arguments which depends on the operation being performed.
-For multi-valued operations the function is only called once, and must evaluate the entire operation as a whole.
+Filter-level access control lets you control which operations can be performed by providing a function which returns a GraphQL filter.
 
-If the function returns:
+- For **mutations**, access will be denied if the item being operated on doesn't match the filter.
+- For **queries**, the access control filter will be combined with the provided query filter and only items which match both filters will be returned.
 
-- `false` then an `Access Denied` error will be returned.
-- `true` then the operation will be allowed.
-- **a declarative value** then this will be applied to the operation as described [above](#declarative-list).
+In general, the filter access control functions will return GraphQL filters.
+They can also return boolean values `true` or `false` to match or exclude all items.
 
 ```typescript
 import { config, createSchema, list } from '@keystone-next/keystone';
+import { checkbox } from '@keystone-next/keystone/fields';
 
 export default config({
   lists: createSchema({
     ListKey: list({
-      // Imperative access control definition
-      access: args => true,
+      fields: {
+        isReadable: checkbox(),
+        isUpdatable: checkbox(),
+        isDeletable: checkbox(),
+      }
+      access: {
+        filter: {
+          query: ({ session, context, listKey, operation }) => {
+            return { isReadable: { equals: true } };
```

Comment:
I'm assuming that the reason that this is not the `true | false` dichotomy of the others is that we add this to the graphql query. Can we call this out a bit more explicitly?

Reply:
Updated to read as such:
```
- For **mutations**, the access control filter will be combined with the unique identifier provided to the operation, and access will be denied if no item is found with this combined filter.
- For **queries**, the access control filter will be combined with the query filter and only items which match both filters will be returned.
```